apps/explorer/priv/repo/migrations/20220315145714_block_number_and_topic_on_events.exs

```diff
@@ -0,0 +1,57 @@
+defmodule Explorer.Repo.Migrations.BlockNumberAndTopicOnEvents do
+  use Ecto.Migration
+  import Ecto.Query
+
+  def up do
+    # add new fields
+    alter table(:celo_contract_events) do
+      add(:block_number, :integer)
+      add(:topic, :string)
+    end
+
+    # assert change is applied to db
+    flush()
+
+    # get block_number and event topic from "parent" log row and update existing event rows
+    from(event in "celo_contract_events",
+      join: log in "logs",
```

Comment:
Why not just join on `blocks`?

Reply:
That would work

Reply:
Oh wait no, we need the `first_topic` from the `logs` entry and joining against `logs` has both block number and topic

Reply:
Ah, it makes sense. I was worrying that while `up` works fine as `block_hash, index` constitute the primary key on the `logs` table but there could be problems with `down` as there is a different index `block_number DESC, index DESC` which might not be used efficiently in that query.